solvebio/resource/dataset.py

```diff
@@ -22,23 +24,68 @@ class Dataset(CreateableAPIResource,
               UpdateableAPIResource):
     """
     Datasets are access points to data. Dataset names are unique
-    within versions of a depository.
+    within a vault folder.
     """
+    RESOURCE_VERSION = 2
+
     LIST_FIELDS = (
-        ('full_name', 'Name'),
-        ('depository', 'Depository'),
-        ('title', 'Title'),
+        ('id', 'ID'),
+        ('vault_name', 'Vault Name'),
+        ('vault_object_path', 'Vault Object Path'),
+        ('vault_object_filename', 'Vault Object Filename'),
+        ('created_at', 'Created'),
         ('description', 'Description'),
     )
 
     @classmethod
-    def get_or_create_by_full_name(cls, full_name, **kwargs):
-        from solvebio import Depository
-        from solvebio import DepositoryVersion
+    def make_full_path(cls, vault_name, path, name):
+        from solvebio import SolveError
+
+        try:
+            user = client.get('/v1/user', {})
+            domain = user['account']['domain']
+        except SolveError as e:
+            print("Error obtaining account domain: {0}".format(e))
+            raise
+
+        result = ':'.join([
+            domain,
+            vault_name,
+            os.path.join('/' + path.lstrip('/'), name),
+        ])
+
+        return result
+
+    @classmethod
+    def get_by_full_path(cls, full_path, **kwargs):
```

Comment:
`retrieve_by_path` feels too long to me, since then we have `retrieve_or_create_by_path`.  

I renamed these all to `get_by...` and `retrieve` will just be used for ID-based grabbing.